### PR TITLE
FEATURE: Use the new plugin show route from the plugin list

### PR DESCRIPTION
**Pull request by martin-brennan**
Followup 4e7a75a7ece3205ce9f3f188b5e016bf75a869c0

Several plugins (Gamification, AI) now use the new
plugin show route. Any plugins that are using it can
now redirect to this page via the Settings button in
the plugin list, rather than taking the user to the
old site settings page filtered by category.

image

### CODE REVIEW

**Comment by martin-brennan** [comment]
Actually I am going to add a component test here too.

**Review by tgxworld — APPROVED** [review]
Code looks OK but I personally will prefer a simple system test to cover this as it prevents features from regressing in the future.

**Comment by martin-brennan** [comment]
@tgxworld system test is tricky because it involves plugins -- I will add the component test for now and add the system test in one of the plugins that relies on this behaviour

File: app/assets/javascripts/discourse/tests/integration/components/admin-plugins-list-item-test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        use_new_show_route: false,
+        full_location: "admin",
+      },
+      has_settings: true,
+    };
+  }
+
+  test("settings link route", async function (assert) {
+    this.currentUser.admin = true;
+    this.plugin = AdminPlugin.create(pluginAttrs());
```
Comment (CvX) [inline app/assets/javascripts/discourse/tests/integration/components/admin-plugins-list-item-test.js:26]: It's preferred to use the store to create new model instances:

```suggestion
    const store = getOwner(this).lookup("service:store");
    this.plugin = store.createRecord("admin-plugin", pluginAttrs());
```

(currently those are equivalent but the `.create()` requires ugly discourse-specific hacks under the hood which we'd like to remove in the future)

Comment (martin-brennan) [reply app/assets/javascripts/discourse/tests/integration/components/admin-plugins-list-item-test.js:26]: @CvX I'm not following sorry, how does looking up service:store work here? And in this case `AdminPlugin.create` is a normal static class helper function that calls the constructor like this:

https://github.com/discourse/discourse/blob/3555ca0965e4b50398fd0d52544a2a809e0dfd1e/app/assets/javascripts/admin/addon/models/admin-plugin.js#L5-L30

Comment (CvX) [reply app/assets/javascripts/discourse/tests/integration/components/admin-plugins-list-item-test.js:26]: Looking up the service is done just like that. Import the function:

```js
import { getOwner } from "@ember/owner";
```

and then in tests use it to look up the store:

```js
const store = getOwner(this).lookup("service:store");
```

`this` in integration/acceptance tests is an object which owner is an application instance.
And the store looks up the class based on the model name you provide:

```js
this.plugin = store.createRecord("admin-plugin", { commit_url: "test" });
this.plugin.commitUrl === "test";
```

**Comment by martin-brennan** [comment]
Ah sorry I see now, I totally glossed over the “store.createRecord” part
😅. Thanks, will fix.

**Pull request merged**